hugo/assets/scripts/arithmetics.tsx

```diff
@@ -0,0 +1,99 @@
+import { MonacoEditorReactComp, monaco, addMonacoStyles } from '@typefox/monaco-editor-react/bundle';
+import { buildWorkerDefinition } from 'monaco-editor-workers';
+import React from 'react';
+import { createRoot } from 'react-dom/client';
+
+buildWorkerDefinition(
+  "../../libs/monaco-editor-workers/workers",
+  new URL("", window.location.href).href,
+  false
+);
+addMonacoStyles('monaco-editor-styles');
+ 
+const syntaxHighlighting = {
+    keywords: [
+        'def','module'
+    ],
+    operators: [
+        '-',',',';',':','*','/','+'
+    ],
+    symbols:  /-|,|;|:|\(|\)|\*|\/|\+/,
+
+    tokenizer: {
+        initial: [
+            { regex: /[_a-zA-Z][\w_]*/, action: { cases: { '@keywords': {"token":"keyword"}, '@default': {"token":"ID"} }} },
+            { regex: /[0-9]+(\.[0-9]*)?/, action: {"token":"number"} },
+            { include: '@whitespace' },
+            { regex: /@symbols/, action: { cases: { '@operators': {"token":"operator"}, '@default': {"token":""} }} },
+        ],
+        whitespace: [
+            { regex: /\s+/, action: {"token":"white"} },
+            { regex: /\/\*/, action: {"token":"comment","next":"@comment"} },
+            { regex: /\/\/[^\n\r]*/, action: {"token":"comment"} },
+        ],
+        comment: [
+            { regex: /[^\/\*]+/, action: {"token":"comment"} },
+            { regex: /\*\//, action: {"token":"comment","next":"@pop"} },
+            { regex: /[\/\*]/, action: {"token":"comment"} },
+        ],
+    }
+} as monaco.languages.IMonarchLanguage;
+
+
+class App extends React.Component<{}, { hasError }> {
```

Comment:
I think you are using React in a good way. I would recommend to use interfaces for generic type parameters. It happens pretty fast that you have to reuse them.